crates/oxc_linter/src/rules/oxc/ifs_same_conditions.rs

```diff
@@ -0,0 +1,191 @@
+use oxc_ast::{
+    ast::{Expression, IfStatement, LogicalOperator, Statement},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+use oxc_span::cmp::ContentEq;
+use oxc_span::{GetSpan, Span};
+
+fn ifs_same_conditions_diagnostic(span1: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Unreachable if consequent")
+        .with_help(
+            "This if statement test will always be false, as the previous `if` statement covers the same condition.",
+        )
```

Comment:
i expanded on clippy's behaviour a decent amount to catch more edge cases.
maybe we rename this rule something like `no-unreachable-else-if` or something like that?
i think these help diagnostic msg could be improved also. but it's late and my brain is stopping working 🙂